skaffold/src/test/kotlin/com/google/container/tools/skaffold/run/SkaffoldFilesComboBoxTest.kt

```diff
@@ -0,0 +1,117 @@
+import io.mockk.impl.annotations.MockK
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/** Unit tests for [SkaffoldFilesComboBox] */
+class SkaffoldFilesComboBoxTest {
+    @get:Rule
+    val containerToolsRule = ContainerToolsRule(this)
+
+    @MockK
+    @TestService
+    lateinit var mockSkaffoldFileService: SkaffoldFileService
+
+    private lateinit var skaffoldFilesComboBox: SkaffoldFilesComboBox
+
+    @Before
+    fun setUp() {
+        skaffoldFilesComboBox = SkaffoldFilesComboBox()
+    }
+
+    @Test
+    fun `empty combobox for empty project has no files`() {
+        val project = containerToolsRule.ideaProjectTestFixture.project
+        every { mockSkaffoldFileService.findSkaffoldFiles(project) } returns listOf()
+        skaffoldFilesComboBox.setProject(project)
+
+        assertThat(skaffoldFilesComboBox.getSelectedSkaffoldFile()).isNull()
+    }
+
+    @Test
+    fun `single skaffold file from project is pre-selected in combo box`() {
+        val project = containerToolsRule.ideaProjectTestFixture.project
+        val mockSkaffoldFile = MockVirtualFile.file("skaffold.yaml")
+        every { mockSkaffoldFileService.findSkaffoldFiles(project) } returns
+            listOf(mockSkaffoldFile)
+        skaffoldFilesComboBox.setProject(project)
+
+        assertThat(skaffoldFilesComboBox.getSelectedSkaffoldFile()).isEqualTo(mockSkaffoldFile)
+    }
+
+    @Test
+    fun `first skaffold file from multi-file project is pre-selected in combo box`() {
+        val project = containerToolsRule.ideaProjectTestFixture.project
+        val mockSkaffoldFile1 = MockVirtualFile.file("k8s/deploy.yaml")
+        val mockSkaffoldFile2 = MockVirtualFile.file("skaffold.yaml")
+        every { mockSkaffoldFileService.findSkaffoldFiles(project) } returns listOf(
+            mockSkaffoldFile1,
+            mockSkaffoldFile2
+        )
+        skaffoldFilesComboBox.setProject(project)
+
+        assertThat(skaffoldFilesComboBox.getSelectedSkaffoldFile()).isEqualTo(mockSkaffoldFile1)
+    }
+
+    @Test
+    fun `setSelectedSkaffoldFile with existing file changes selection in combo box`() {
+        val project = containerToolsRule.ideaProjectTestFixture.project
+        val mockSkaffoldFile1 = MockVirtualFile.file("k8s/deploy.yaml")
+        val mockSkaffoldFile2 = MockVirtualFile.file("skaffold.yaml")
+        every { mockSkaffoldFileService.findSkaffoldFiles(project) } returns listOf(
+            mockSkaffoldFile1,
+            mockSkaffoldFile2
+        )
+        skaffoldFilesComboBox.setProject(project)
+        skaffoldFilesComboBox.setSelectedSkaffoldFile(mockSkaffoldFile2)
+
+        // make sure the file is not double-added.
+        assertThat(skaffoldFilesComboBox.model.size).isEqualTo(2)
+        assertThat(skaffoldFilesComboBox.getSelectedSkaffoldFile()).isEqualTo(mockSkaffoldFile2)
+    }
+
+    @Test
+    fun `setSelectedSkaffoldFile with non-existing file changes selection and adds element`() {
```

Comment:
Why do we want this behavior? Seems to me that if you try to set a non-existent skaffold file then it should be an error.

Reply:
For existing run configurations where Skaffold file was set. If user opens them and that file was removed, I believe we still want this selection to show (and remind what was the configuration before), but the error will show as well saying we don't have this file in the project anymore.

Reply:
ah. yea makes sense.